### ISSUE TITLE
React Native S3 image upload returns “Stream Closed” using XHR

### ISSUE THREAD

**Issue by usama-asfar** [facebook/react-native#29586]
After updating React Native version to latest 0.63.2 and trying to upload the image to S3 bucket XHR returns error Stream Closed image upload was working fine with version 0.61.5

The Code

```js
  const xhr = new XMLHttpRequest();
        xhr.open('PUT', presignedUrl);
        xhr.setRequestHeader('Content-Type', mime);
        xhr.send({ uri: path, type: mime });

        xhr.onreadystatechange = function () {
          if (xhr.readyState === 4) {
            if (xhr.status === 200) {
              console.log('Image successfully uploaded to S3');
            } else {
              console.log(
                'Error while sending the image to S3',
                xhr.responseText,
                xhr.response,
              );
            }
          }
        };
```
When the user wants to upload a profile image first App send a request to the server and get return the pre-signed URL and upload from client-side this how App was working.

dev env info
```
System:
    OS: Windows 10 10.0.19041
    CPU: (4) x64 Intel(R) Core(TM) i3-2377M CPU @ 1.50GHz
    Memory: 8.45 GB / 11.80 GB
  Binaries:
    Node: 12.18.3 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 6.14.6 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.1
      System Images: android-27 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6626763
  Languages:
    Java: 1.8.0_211
    Python: 3.8.5
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.2 => 0.63.2
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by usama-asfar** [comment]
This is not an S3 issue this issue with react native latest version `0.63.2` when using XHR

**Comment by usama-asfar** [comment]
solved #28551 at [link](https://github.com/facebook/react-native/issues/28551#issuecomment-611085378)
